exploration-module/src/test/java/io/edpn/backend/exploration/adapter/persistence/CreateSystemPortTest.java

```diff
@@ -0,0 +1,77 @@
+package io.edpn.backend.exploration.adapter.persistence;
+
+import io.edpn.backend.exploration.adapter.persistence.entity.SystemEntity;
+import io.edpn.backend.exploration.application.domain.System;
+import io.edpn.backend.exploration.application.dto.mapper.SystemEntityMapper;
+import io.edpn.backend.exploration.application.port.outgoing.CreateSystemPort;
+import io.edpn.backend.util.IdGenerator;
+import io.edpn.backend.util.exception.DatabaseEntityNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class CreateSystemPortTest {
+    @Mock
+    private MybatisSystemRepository mybatisSystemRepository;
+
+    @Mock
+    private SystemEntityMapper systemEntityMapper;
+
+    @Mock
+    private IdGenerator idGenerator;
+
+    private CreateSystemPort underTest;
+
+    @BeforeEach
+    void setUp() {
+        underTest = new SystemRepository(mybatisSystemRepository, systemEntityMapper, idGenerator);
+    }
+
+    @Test
+    void create_shouldInsertAndLoad() {
+        // Given
```

Comment:
From the department of potentially unpopular opinions: I think religiously repeating the comments "given - when - then" is just noise. Tests pretty much always have the sequence of setting up, performing the action, and verifying the results, and hence this requires no comments to understand.